File: UIPeriodicTable/CPlusPlus/Resources/Components/code_uimessagewindow.cpp
// UIMessageWindow application source code
#include <Atomic/UI/UIMessageWindow.h>
#include "PeriodicApp.h"

void PeriodicApp::setup_uimessagewindow( UIWidget *layout )
{
    PODVector<UIWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp, HandleUimessagewindowEvent ));
}

void PeriodicApp::HandleUimessagewindowEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    UIWidget* widget = static_cast<UIWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    UIWidget* mylayout = static_cast<UIWidget*>(widget->FindWidget("pageuimessagewindow"));
    String refid = eventData[P_REFID].GetString();
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uimessagewindowcode" )
        {
            AppLog( "UIMessageWindow support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Components/code_uimessagewindow.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uimessagewindowlayout" )
        {
            AppLog( "UIMessageWindow support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_uimessagewindow.ui.txt", widget->GetParent() );
        }

        if (widget->GetId() ==  "msgnone" )
        {
            AppLog( "UIMessageWindow support : " + widget->GetId() + " was pressed ");
            UIMessageWindow *mess1 = new UIMessageWindow(context_, mylayout, "mymess1");
            mess1->Show("MessageWindow - NONE", "this is a MessageWindow - None button", (UI_MESSAGEWINDOW_SETTINGS)0, 0, 0, 0);
        }
        if (widget->GetId() ==  "msgok" )
        {
            AppLog( "UIMessageWindow support : " + widget->GetId() + " was pressed ");
            UIMessageWindow *mess2 = new UIMessageWindow(context_, mylayout, "mymess2");
            SubscribeToEvent(mess2, E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp, HandleUimessagewindowEvent ));
            mess2->Show("MessageWindow - OK", "this is a MessageWindow - OK button", UI_MESSAGEWINDOW_SETTINGS_OK, 0, 0, 0);
        }
        if (widget->GetId() ==  "msgkcancel" )
        {
            AppLog( "UIMessageWindow support : " + widget->GetId() + " was pressed ");
            UIMessageWindow *mess3 = new UIMessageWindow(context_, mylayout, "mymess3");
            SubscribeToEvent(mess3, E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp, HandleUimessagewindowEvent ));
            mess3->Show("MessageWindow - OK CANCEL", "this is a MessageWindow - OK CANCEL buttons", UI_MESSAGEWINDOW_SETTINGS_OK_CANCEL, 0, 0, 0);
        }
        if (widget->GetId() ==  "msgyesno" )
        {
            AppLog( "UIMessageWindow support : " + widget->GetId() + " was pressed ");
            UIMessageWindow *mess4 = new UIMessageWindow(context_, mylayout, "mymess4");
            SubscribeToEvent(mess4, E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp, HandleUimessagewindowEvent ));
            mess4->Show( "MessageWindow - YES NO", "this is a MessageWindow - YES NO buttons", UI_MESSAGEWINDOW_SETTINGS_YES_NO, 0, 0, 0);
        }

        if (refid ==  "TBMessageWindow.ok" )
        {
            AppLog( "UIMessageWindow event : " + refid + " closed the UIMessageWindow");
        }
        if (refid ==  "TBMessageWindow.cancel" )
        {
            AppLog( "UIMessageWindow event : " + refid + " closed the UIMessageWindow");
        }
        if (refid ==  "TBMessageWindow.yes" )
        {
            AppLog( "UIMessageWindow event : " + refid + " closed the UIMessageWindow");
        }
        if (refid ==  "TBMessageWindow.no" )
        {
            AppLog( "UIMessageWindow event : " + refid + " closed the UIMessageWindow");
        }
    }
    else
    {
        AppLog( "UIMessageWindow event : " + widget->GetId() + " event type = " + EventReport(eventData[P_TYPE].GetInt()));
    }
}

File: UIPeriodicTable/CSharp/Resources/Components/code_uimessagewindow.cs
// UIMessageWindow application source code
using System;
using AtomicEngine;

public class code_uimessagewindow : CSComponent {

    public void Setup( UIWidget  layout )
    {
        var dest = new AtomicEngine.Vector<AtomicEngine.UIWidget>();
        layout.SearchWidgetClass( "TBButton", dest );
        for (var ii = 0; ii < dest.Size; ii++) {
            dest[ii].SubscribeToEvent<WidgetEvent> (dest [ii], HandleUimessagewindowEvent );
        }
    }

    private static void HandleUimessagewindowEvent ( WidgetEvent ev )
    {
        UIWidget widget = (UIWidget)ev.Target;
        string refid  = (string)ev.RefID;
        if ( widget.Equals(null)) return;
        UIWidget  mylayout = (UIWidget)widget.FindWidget("pageuimessagewindow");
        if ( ev.Type == UI_EVENT_TYPE.UI_EVENT_TYPE_CLICK) {
            if (widget.GetId() == "uimessagewindowcode" ) {
                AtomicMain.AppLog( "UIMessageWindow support : " + widget.GetId() + " was pressed " );
                AtomicMain.ViewCode ( "Components/code_uimessagewindow.cs", widget.GetParent() );
            }
            if (widget.GetId() ==  "uimessagewindowlayout" ) {
                AtomicMain.AppLog( "UIMessageWindow support : " + widget.GetId() + " was pressed ");
                AtomicMain.ViewCode ( "Scenes/layout_uimessagewindow.ui.txt", widget.GetParent() );
            }

            if (widget.GetId() ==  "msgnone" ) {
                AtomicMain.AppLog( "UIMessageWindow support : " + widget.GetId() + " was pressed ");
                UIMessageWindow  mess1 = new UIMessageWindow( mylayout, "mymess1");
                mess1.Show("MessageWindow - NONE", "this is a MessageWindow - None button", (UI_MESSAGEWINDOW_SETTINGS)0, false, 0, 0);
            }
            if (widget.GetId() ==  "msgok" ) {
                AtomicMain.AppLog( "UIMessageWindow support : " + widget.GetId() + " was pressed ");
                UIMessageWindow  mess2 = new UIMessageWindow( mylayout, "mymess2");
                mess2.SubscribeToEvent<WidgetEvent> (mess2, HandleUimessagewindowEvent );
                mess2.Show("MessageWindow - OK", "this is a MessageWindow - OK button", UI_MESSAGEWINDOW_SETTINGS.UI_MESSAGEWINDOW_SETTINGS_OK, false, 0, 0);
            }
            if (widget.GetId() ==  "msgkcancel" ) {
                AtomicMain.AppLog( "UIMessageWindow support : " + widget.GetId() + " was pressed ");
                UIMessageWindow  mess3 = new UIMessageWindow( mylayout, "mymess3");
                mess3.SubscribeToEvent<WidgetEvent> (mess3, HandleUimessagewindowEvent );
                mess3.Show("MessageWindow - OK CANCEL", "this is a MessageWindow - OK CANCEL buttons", UI_MESSAGEWINDOW_SETTINGS.UI_MESSAGEWINDOW_SETTINGS_OK_CANCEL, false, 0, 0);
            }
            if (widget.GetId() ==  "msgyesno" ) {
                AtomicMain.AppLog( "UIMessageWindow support : " + widget.GetId() + " was pressed ");
                UIMessageWindow  mess4 = new UIMessageWindow( mylayout, "mymess4");
                mess4.SubscribeToEvent<WidgetEvent> (mess4, HandleUimessagewindowEvent );
                mess4.Show( "MessageWindow - YES NO", "this is a MessageWindow - YES NO buttons", UI_MESSAGEWINDOW_SETTINGS.UI_MESSAGEWINDOW_SETTINGS_YES_NO, false, 0, 0);
            }

            if (refid ==  "TBMessageWindow.ok" ) {
                AtomicMain.AppLog( "UIMessageWindow event : " + refid + " closed the UIMessageWindow");
            }
            if (refid ==  "TBMessageWindow.cancel" ) {
                AtomicMain.AppLog( "UIMessageWindow event : " + refid + " closed the UIMessageWindow");
            }
            if (refid ==  "TBMessageWindow.yes" ) {
                AtomicMain.AppLog( "UIMessageWindow event : " + refid + " closed the UIMessageWindow");
            }
            if (refid ==  "TBMessageWindow.no" ) {
                AtomicMain.AppLog( "UIMessageWindow event : " + refid + " closed the UIMessageWindow");
            }
        } else {
            AtomicMain.AppLog( "UIMessageWindow event : " + widget.GetId() + " event type = " + AtomicMain.EventReport((int)ev.Type));
        }
    }
}

